File: Game/src/main/java/net/winternetwork/bedwars/game/modules/shop/model/VillagerModel.java
package net.winternetwork.bedwars.game.modules.shop.model;

import lombok.SneakyThrows;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.entity.Entity;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Villager;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;


public class VillagerModel {

    @SneakyThrows
    public static Villager spawnVillager(Location location) {
        World world = location.getWorld();

        Villager villager = ((Villager) world.spawnEntity(location, EntityType.VILLAGER));
        villager.setCustomNameVisible(true);
        villager.setCustomName("§e§lLOJA");

        setAi(villager, false);

        return villager;
    }

    private static void setAi(Entity entity, boolean enabled) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, ClassNotFoundException, InstantiationException {
        Method handleMethod = entity.getClass().getMethod("getHandle");
        Object nmsEntity = handleMethod.invoke(entity);

        Method tagMethod = nmsEntity.getClass().getMethod("getNBTTag");
        Object tag = tagMethod.invoke(nmsEntity);

        if (tag == null)
            tag = Class.forName("net.minecraft.server.v1_8_R3.NBTTagCompound").newInstance();

        Method cMethod = nmsEntity.getClass().getMethod("c");
        cMethod.invoke(nmsEntity, tag);

        Method setIntMethod = tag.getClass().getMethod("setInt");
        int value = enabled ? 0 : 1;
        setIntMethod.invoke(tag, "NoAI", value);

        Method fMethod = nmsEntity.getClass().getMethod("f");
        fMethod.invoke(nmsEntity, tag);
    }
}

File: Game/src/main/kotlin/net/winternetwork/bedwars/modules/shop/model/VillagerModel.kt
package net.winternetwork.bedwars.modules.shop.model

import org.bukkit.Location
import org.bukkit.entity.Entity
import org.bukkit.entity.EntityType
import org.bukkit.entity.Villager

object VillagerModel {

    fun spawnVillager(location: Location): Villager {
        return (location.world.spawnEntity(location, EntityType.VILLAGER) as Villager)
                .apply {
                    isCustomNameVisible = true
                    customName = "§e§lLOJA"

                    setAi(false)
                }
    }

    private fun Entity.setAi(enabled: Boolean) {
        val handleMethod = javaClass.getMethod("getHandle")
        val entity = handleMethod.invoke(this)

        val tagMethod = entity.javaClass.getMethod("getNBTTag")
        val tag = tagMethod.invoke(entity)
                ?: Class.forName("net.minecraft.server.v1_8_R3.NBTTagCompound")
                        .newInstance()

        entity.javaClass.getMethod("c")
                .invoke(entity, tag)

        val value = if (enabled) 0 else 1
        tag.javaClass.getMethod("setInt")
                .invoke(tag, "NoAI", value)

        entity.javaClass.getMethod("f")
                .invoke(entity, tag)
    }
}